pkg/wasm/httpfetcher.go

```diff
@@ -25,37 +27,43 @@ import (
 
 // HTTPFetcher fetches remote wasm module with HTTP get.
 type HTTPFetcher struct {
-	defaultClient  *http.Client
+	client         *http.Client
+	insecureClient *http.Client
 	initialBackoff time.Duration
 }
 
 // NewHTTPFetcher create a new HTTP remote wasm module fetcher.
 func NewHTTPFetcher() *HTTPFetcher {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	return &HTTPFetcher{
-		defaultClient: &http.Client{
-			Timeout: 5 * time.Second,
+		client: &http.Client{},
+		insecureClient: &http.Client{
+			Transport: transport,
 		},
 		initialBackoff: time.Millisecond * 500,
 	}
 }
 
 // Fetch downloads a wasm module with HTTP get.
-func (f *HTTPFetcher) Fetch(url string, timeout time.Duration) ([]byte, error) {
-	c := f.defaultClient
-	if timeout != 0 {
-		c = &http.Client{
-			Timeout: timeout,
-		}
+func (f *HTTPFetcher) Fetch(ctx context.Context, url string, allowInsecure bool) ([]byte, error) {
+	c := f.client
+	if allowInsecure {
+		c = f.insecureClient
 	}
 	attempts := 0
-
 	b := backoff.NewExponentialBackOff()
 	b.InitialInterval = f.initialBackoff
 	b.Reset()
 	var lastError error
 	for attempts < 5 {
 		attempts++
-		resp, err := c.Get(url)
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```

Comment:
nit: More grace way is to check the attempts times and ctx.Done simultaneously

Reply:
Or I guess the easier way is to set the Client.Timeout = 10s

Reply:
Good catch! I have added that.